llvm/tools/llvm-exegesis/lib/BenchmarkRunner.cpp

```diff
@@ -317,6 +317,7 @@ class SubProcessFunctionExecutorImpl
     int CounterFileDescriptor = Counter->getFileDescriptor();
     Error SendError =
         sendFileDescriptorThroughSocket(WriteFD, CounterFileDescriptor);
+    close(WriteFD);
```

Comment:
should we use `make_scope_exit` to make sure this fd is closed no matter how we leave this function?

Reply:
Good point. I've used it within all of these functions to ensure that the file descriptor is properly closed upon exiting the function (especially with error conditions). Thanks for pointing this out! I hadn't heard about `make_scope_exit` before.